auto_submit/lib/service/validation_service.dart

```diff
@@ -79,19 +79,66 @@ class ValidationService {
     await processPullRequest(config, queryResult, messagePullRequest, ackId, pubsub);
   }
 
-  /// Processes a PullRequest running several validations to decide whether to
-  /// land the commit or remove the autosubmit label.
+  /// Processes a PullRequest.
+  ///
+  /// If it is a ToT revert, the PR will be landed without validations.
+  /// If it is not a ToT revert, the PR will be landed if all validations pass.
   Future<void> processPullRequest(
       Config config, QueryResult result, github.PullRequest messagePullRequest, String ackId, PubSub pubsub) async {
+    github.RepositorySlug slug = messagePullRequest.base!.repo!.slug();
+    final PullRequest pullRequest = result.repository!.pullRequest!;
+    Commit commit = pullRequest.commits!.nodes!.single.commit!;
+    final String? sha = commit.oid;
+    final GithubService gitHubService = await config.createGithubService(slug);
+    final bool isTotRevert = await checkIsTotRevert(sha!, slug, gitHubService);
+    bool shouldMergePR = false;
+    if (!isTotRevert) {
+      shouldMergePR = await checkShoudMergePR(config, result, messagePullRequest, ackId, pubsub, slug, gitHubService);
+      if (!shouldMergePR) {
+        return;
+      }
+    }
+    // If we got to this point it means we are ready to submit the PR.
+    bool processed = await processMerge(config, result, messagePullRequest, sha);
+    if (processed) {
+      await pubsub.acknowledge('auto-submit-queue-sub', ackId);
+      log.info('Acked the processed message : $ackId.');
+    }
+  }
+
+  /// Check if the `commitSha` is a clean revert of TOT commit.
```

Comment:
This should allow reverting anything within the past 10-15 commits or so since several commits could be landed on top of the bad commit that needs to be reverted.

Reply:
That functionality will be implemented once we have a revert API, github does not provide a way to identify a PR is a revert.

Reply:
As written this PR is not useful. It is very rare that the commit that needs to be reverted is directly at the top of the tree. As it turns out, at this moment, the tree is closed on the second commit back, and the revert is being blocked waiting for review.

Reply:
@ricardoamador @keyonghan in terms of effort how difficult would it be to auto-approve the PR if it was created by a member of `flutter-hackers` and it contains the `warning: land on red to fix tree breakage` label?

if that functionality is implemented would we need the revert API?

Reply:
I would think it would be fairly simple to integrate this into the autosubmit bot by adding a user membership check. We already are looking for that label. I think this would be the functionality for the revert api. I would like @keyonghan to weight in to though.

Reply:
Currently we use `warning: land on red to fix tree breakage` only to ignore tree status.
I guess the purpose of this label is different from what we are targeting here: for PRs that intend to unblock the tree may need pre-submit check validation, and then get landed by ignoring the tree status.
Here we may need a new label. With that, it will get approval and auto landed immediately without waiting for any check/status.
I believe the efforts should be straightforwards based on existing validation workflows: 
1) add new label check when auto approving
2) ignore statuses/checks in CI validation

One thing to note is the autosubmit app is listening to `autosubmit` label, there might be some race condition when adding the new label.